processor/src/main/java/org/mapstruct/ap/internal/model/source/MappingOptions.java

```diff
@@ -178,11 +181,47 @@ public void applyInheritedOptions(MappingOptions inherited, boolean isInverse, S
                 }
             }
 
+
             // now add all of its own mappings
             newMappings.putAll( getMappings() );
+
+            // filter new mappings
+            filterNestedTargetIgnores( newMappings );
+
             setMappings( newMappings );
         }
 
         markAsFullyInitialized();
     }
+
+    private void filterNestedTargetIgnores( Map<String, List<Mapping>> mappings) {
+
+        // collect all properties to ignore, and safe their target name ( == same name as first ref target property)
+        Set<String> ignoreList = new HashSet<String>();
```

Comment:
rename this `ignoreList` and the `removeList` below to `ignored` / `toBeRemoved` or something? As they are not really lists... :)